client_test.go

```diff
@@ -222,6 +224,28 @@ func TestS3PointerUnmarshalInvalidLength(t *testing.T) {
 	assert.ErrorContains(t, err, "invalid pointer format, expected length 2, but received [3]")
 }
 
+func TestWithObjectPrefix(t *testing.T) {
+	invalidPrefixes := []string{"../test", "./test", "tes&", "te$t", "testñ", "te@st", "test=", "test;", "test:", "+test", "te st", "te,st", "test?"}
```

Comment:
There a way in Go to write tests that run multiple scenarios. These are usually called table-driven unit tests:
https://yourbasic.org/golang/table-driven-unit-test/

You may want to check it out because it could simplify your code here a bit. There are also many places in our own code in highlander where we do this. Just to mention an example, check out the `middleware_test.go`  for many examples of the pattern.

Reply:
In this case, I'm not using table-driven unit tests because the current tests doesn't do it like that. So I'm not sure if we want to introduce table-driven unit tests or continue with how the maintainer have it.

Do you think that we should introduce that way to do the tests?

Reply:
You can consider my comment on this test as nitpicking. No need to do it if you don't think it necessary. That being said, because the maintainer is not doing it, is not a reason to avoid it. I was just thinking that there seems to be two obvious scenarios in your code that one may want to run and debug separately.